python/cudf/cudf/core/column/column.py

```diff
@@ -2004,7 +2004,11 @@ def as_column(
         if dtype is not None:
             data = data.astype(dtype)
 
-    elif is_scalar(arbitrary) and not isinstance(arbitrary, memoryview):
+    elif (
+        is_scalar(arbitrary)
+        and not isinstance(arbitrary, memoryview)
+        and not _is_categorical_dtype(dtype)
```

Comment:
Would slightly prefer to include this in the `if isinstance(arbitrary, pd.Interval):` branch as below like `if isinstance(arbitrary, pd.Interval) or  _is_categorical_dtype(dtype)`

Reply:
Done 👍